### PR TITLE
Update based on MSVC static analyzer

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
These issues were found while consuming the library in dotnet/runtime.
- https://github.com/dotnet/runtime/pull/66234
- https://github.com/dotnet/runtime/pull/66427
- https://github.com/dotnet/runtime/pull/66676

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@djwatson or another person with maintainer rights. Not trying to pressure on getting this merged, but a quick read on commit https://github.com/libunwind/libunwind/pull/333/commits/0b90eea4d8565f863c2084ddddef9986acf96105 would be helpful. Is there a preference for one of the following "negating" of an unsigned type:

1) `~n + 1` : formal definition for negation in two-complement logic
2) `(US_T)(-(S_T)n)` : read as cast `n` to the signed type (S_T), negate, and then recast back to the unsigned type (US_T).
3) `n & ~(sizeof(T) - 1)` : Mentioned at https://github.com/dotnet/runtime/pull/66427#issuecomment-1063605812

The current PR uses option (1), but there has been some question about whether (2) or (3) might be more appropriate.

**Comment by djwatson** [comment]
I don't really have a strong preference

**Comment by djwatson** [comment]
Actually - for whichever method, it might be more clear to make a macro out of it.

**Comment by AaronRobinsonMSFT** [comment]
Thanks @djwatson. I found an existing macro to use, `UNW_ALIGN`, and used casting when it seemed appropriate.

File: src/dwarf/Gparser.c [hunk 1]
```diff
@@ -390,9 +390,9 @@ run_cfi_program (struct dwarf_cursor *c, dwarf_state_record_t *sr,
           if (((ret = read_regnum (as, a, addr, &regnum, arg)) < 0)
               || ((ret = dwarf_read_uleb128 (as, a, addr, &val, arg)) < 0))
             break;
-          set_reg (sr, regnum, DWARF_WHERE_CFAREL, -(val * dci->data_align));
+          set_reg (sr, regnum, DWARF_WHERE_CFAREL, (unw_word_t)(-(unw_sword_t)val));
           Debug (15, "CFA_GNU_negative_offset_extended cfa+0x%lx\n",
-                 (long) -(val * dci->data_align));
+                 (long)(unw_word_t)(-(unw_sword_t)val));
```
Comment (am11) [inline src/dwarf/Gparser.c:395]: Today the CI is fixed. Looks like Gparser on win-arm64 needs casts in a few more places:  https://app.travis-ci.com/github/libunwind/libunwind/jobs/571010144
```
C:\Users\travis\build\libunwind\libunwind\src\dwarf\Gparser.c(821): error C2146: syntax error: missing ';' before identifier '__asm__' [C:\Users\travis\build\libunwind\libunwind\bin\windows-x64\aarch64-linux-gnu\src\libunwind.vcxproj]

C:\Users\travis\build\libunwind\libunwind\src\dwarf\Gparser.c(821): warning C4013: '__asm__' undefined; assuming extern returning int [C:\Users\travis\build\libunwind\libunwind\bin\windows-x64\aarch64-linux-gnu\src\libunwind.vcxproj]

C:\Users\travis\build\libunwind\libunwind\src\dwarf\Gparser.c(821): warning C4244: '=': conversion from 'unw_word_t' to 'int', possible loss of data [C:\Users\travis\build\libunwind\libunwind\bin\windows-x64\aarch64-linux-gnu\src\libunwind.vcxproj]

C:\Users\travis\build\libunwind\libunwind\src\dwarf\Gparser.c(821): error C2106: '=': left operand must be l-value [C:\Users\travis\build\libunwind\libunwind\bin\windows-x64\aarch64-linux-gnu\src\libunwind.vcxproj]

C:\Users\travis\build\libunwind\libunwind\src\dwarf\Gparser.c(822): error C2143: syntax error: missing ')' before ':' [C:\Users\travis\build\libunwind\libunwind\bin\windows-x64\aarch64-linux-gnu\src\libunwind.vcxproj]
```

Comment (vtjnash) [inline src/dwarf/Gparser.c:395]: This appears also to have been a functional change. Is this correct?

**Comment by AaronRobinsonMSFT** [comment]
@djwatson Looks like I made a mistake here. Let me try this again. Apologies.

File: src/dwarf/Gparser.c (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/dwarf/Gparser.c:395]: I just got back to looking at this... I really must apologize. I've no idea how this happened as the above is clearly not correct.

**Pull request merged**